### PR TITLE
Tickets/dm 13960

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: core/modules/parser/QSMySqlListener.cc [hunk 1]
```diff
@@ -1094,40 +1784,90 @@ class UdfFunctionCallAdapter :
...
+                        QSMySqlParser::ScalarFunctionNameContext* ctx)
+    : AdapterT(parent)
+    , _ctx(ctx) {}
+
+    void handleFunctionNameBase(const string& name) override {
+        _name = name;
+    }
+
+    void onExit() override {
+        string str;
```
Comment (andy-slac) [inline core/modules/parser/QSMySqlListener.cc]: unused?

File: core/modules/parser/QSMySqlListener.cc [hunk 2]
```diff
@@ -1247,6 +2020,56 @@ ostream& operator<<(ostream& os, const LogicalExpressionAdapter& logicalExpressi
...
+    void onExit() {
+        ASSERT_EXECUTION_CONDITION(false == _expressions.empty() && _predicate != nullptr,
+                "InPredicateAdapter was not fully populated:" << *this, _ctx);
+        auto inPredicate = std::make_shared<query::InPredicate>();
+        inPredicate->value = _predicate;
+        inPredicate->cands = _expressions;
+        lockedParent()->handleInPredicate(inPredicate);
+    }
+
+    friend ostream& operator<<(ostream& os, const InPredicateAdapter& inPredicateAdapter) {
```
Comment (andy-slac) [inline core/modules/parser/QSMySqlListener.cc]: Do you need this, you don't seem to have it in other classes?

File: core/modules/parser/QSMySqlListener.cc [hunk 3]
```diff
@@ -1302,22 +2172,78 @@ class UnaryExpressionAtomAdapter :
...
+        _valueFactor = valueFactor;
+    }
+
+    void onExit() override {
+        ASSERT_EXECUTION_CONDITION(false == _operatorPrefix.empty() && _valueFactor != nullptr,
+                "Expected unary operator (" << _operatorPrefix << ") and ValueFactor(" << _valueFactor <<
+                "to be populated", _ctx);
+        ASSERT_EXECUTION_CONDITION(_valueFactor->getType() == query::ValueFactor::CONST,
+                "Currently can only handle const val", _ctx);
+            _valueFactor->setConstVal(_operatorPrefix + _valueFactor->getConstVal());
```
Comment (andy-slac) [inline core/modules/parser/QSMySqlListener.cc]: over-indented?

File: core/modules/parser/WhereFactory.cc [hunk 4]
```diff
@@ -225,9 +226,19 @@ WhereFactory::_import(antlr::RefAST a) {
...
                                  const std::string& function,
-                                 const std::vector<std::string>& parameters) {
+                                 const std::vector<std::shared_ptr<query::ValueFactor>>& parameters) {
+
+    // Here we extract the args from a vector of ValueFactor::ColumnRef
+    // This is a side effect of the current IR, where in most cases a constant string is represented as
+    // a column name. But in a QservRestrictor (aka QservFunction) each par is simply represented by a string.
     auto restrictor = std::make_shared<query::QsRestrictor>();
-    restrictor->_params = parameters;
+    for (auto const valueFactor : parameters) {
```
Comment (andy-slac) [inline core/modules/parser/WhereFactory.cc]: `auto const&`?

File: core/modules/qana/TablePlugin.cc [hunk 5]
```diff
@@ -215,10 +215,10 @@ class fixExprAlias {
     void _patchStar(query::ValueFactor& vt) {
         // TODO: No support for <db>.<table>.* in framework
         // Only <table>.* is supported.
-        std::string newAlias = _getAlias("", vt.getTableStar());
+        std::string newAlias = _getAlias("", vt.getConstVal());
         if (newAlias.empty()) { return; } //  Ignore if no replacement
                                          //  exists.
-        else { vt.setTableStar(newAlias); }
+        else { vt.setConstVal(newAlias); }
```
Comment (andy-slac) [inline core/modules/qana/TablePlugin.cc]: This if/else needs to be re-formatted.

File: core/modules/query/BoolTerm.h [hunk 6]
```diff
@@ -139,13 +139,23 @@ class BoolTerm {
 };
 
 
+class BoolFactor;
+
 
 class LogicalTerm : public BoolTerm {
 public:
-    void addBoolTerm(BoolTerm::Ptr& boolTerm) {
+    void addBoolTerm(BoolTerm::Ptr boolTerm) {
```
Comment (andy-slac) [inline core/modules/query/BoolTerm.h]: `BoolTerm::Ptr const&`?

File: core/modules/query/BoolTerm.h [hunk 7]
```diff
@@ -340,6 +353,9 @@ class BoolTermFactor : public BoolFactorTerm {
 public:
     typedef std::shared_ptr<BoolTermFactor> Ptr;
 
+    BoolTermFactor() {}
+    BoolTermFactor(std::shared_ptr<BoolTerm> term) : _term(term) {}
```
Comment (andy-slac) [inline core/modules/query/BoolTerm.h]: `const&`

File: core/modules/query/GroupByClause.h [hunk 8]
```diff
@@ -60,6 +60,9 @@ class GroupByTerm {
     GroupByTerm() {}
     ~GroupByTerm() {}
 
+    GroupByTerm(ValueExprPtr valueExpr, std::string collate)
```
Comment (andy-slac) [inline core/modules/query/GroupByClause.h]: `const&` x2

File: core/modules/query/ValueFactor.h [hunk 9]
```diff
@@ -106,7 +106,10 @@ class ValueFactor {
     std::shared_ptr<FuncExpr> _funcExpr;
     std::shared_ptr<ValueExpr> _valueExpr;
     std::string _alias;
-    std::string _tableStar; // Reused as const val (no tablestar)
+    std::string _constVal;  // formerly named "tablestar"
+                            // seems to often contain a string representation of a number. Can also contain
+                            // `*` (presumably as in `SELECT *`. It would probably be good to factor it so
+                            // so that `constVal` is exclusivly a string number and `*` is a different field.
```
Comment (andy-slac) [inline core/modules/query/ValueFactor.h]: exclusively

File: core/modules/parser/QSMySqlLexer.g4 [hunk 10]
```diff
@@ -2,9 +2,1031 @@ lexer grammar QSMySqlLexer;
...
+fragment Z:('z'|'Z');
+
+
 // Qserv Function Names:
-QSERV_AREASPEC_BOX:                 'qserv_areaspec_box';
-QSERV_AREASPEC_CIRCLE:              'qserv_areaspec_circle';
-QSERV_AREASPEC_ELLIPSE:             'qserv_areaspec_ellipse'; 
-QSERV_AREASPEC_POLY:                'qserv_areaspec_poly';
-QSERV_AREASPEC_HULL:                'qserv_areaspec_hull'; 
+QSERV_AREASPEC_BOX:                 Q S E R V '_' A R E A S P E C '_' B O X;
```
[the remark this answers is not in the document]
Comment (n8pease) [reply core/modules/parser/QSMySqlLexer.g4]: it's either that or write custom character streams.

[the remark this answers is not in the document]
Comment (n8pease) [reply core/modules/parser/QSMySqlLexer.g4]: https://github.com/antlr/antlr4/pull/2046

[the remark this answers is not in the document]
Comment (n8pease) [reply core/modules/parser/QSMySqlLexer.g4]: after reading this thread https://github.com/antlr/antlr4/pull/2046
I'm fairly convinced doing it this way (in the lexer.g4 file) is the Right Thing. There is a slightly hacky way to do it using a custom `CharStream` that converts everything to upper (or to lower) before comparing, but that is a simplistic approach and becomes problematic with certain use cases.

**Comment by n8pease** [comment]
merged via antlr4 integration branch.

**Pull request closed without merging**